engine/classes/Elgg/HooksRegistrationService.php

```diff
@@ -90,6 +90,33 @@ public function unregisterHandler($name, $type, $callback) {
 	}
 
 	/**
+	 * Removes all handlers
+	 *
+	 * @param string   $name
+	 * @param string   $type
+	 *
+	 * @return array
+	 * @access private
+	 */
+	public function clearHandlers($name, $type = null) {
```

Comment:
We should make it clear that handlers registered for `foo, all` and `all, all` would need to be cleared separately.